### PR TITLE
Implement generic data type

### PR DESCRIPTION
**Pull request by V1talie**

### CODE REVIEW

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
-import java.util.Collection;
+import java.util.*;
 
-public class StudentList implements List<Student> {
+public class StudentList<T> implements List<T> {
 
     private int size;
     private int maxsize;
-    private Student[] temp;
+    private Object[] temp;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: rename plz to a more explicit name like elementArray

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 2]
```diff
@@ -240,9 +243,9 @@ public boolean retainAll(Collection<?> collection) {
         throw new UnsupportedOperationException();
     }
 
-    private class MyIterator implements ListIterator<Student> {
+    private class MyIterator implements ListIterator<T> {
 
-        private int currentPosition;
+        int currentPosition;
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: add private modifier

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 3]
```diff
@@ -192,25 +196,24 @@ public List<Student> subList(int i, int i1) {
...
-    public boolean addAll(Collection<? extends Student> collection) {
+    public boolean addAll(Collection<? extends T> collection) {
         Object[] contain = collection.toArray();
         int leng = contain.length;
         if (leng == 0) return false;
-        Student[] temp = this.temp;
         if (leng > size) {
             growSize(leng);
         }
+        System.out.println(size);
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java]: remove this

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 1 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 2 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java (hunk 3 above)
Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java]: Done

File: src/main/java/com/endava/internship/collections/StudentList.java [hunk 4]
```diff
@@ -96,59 +93,62 @@ public boolean remove(Object o) {
...
-        temp = Arrays.copyOf(temp, maxsize);
+        elementArray = Arrays.copyOf(elementArray, maxsize);
     }
 
     @Override
-    public Student remove(int i) {
-        temp[i] = null;
-        Student oldValue = null;
+    @SuppressWarnings("unchecked")
+    public T remove(int i) {
```
Comment (GAlexandru-be) [inline src/main/java/com/endava/internship/collections/StudentList.java:148]: do a check that index index < 0 and index >= length and throw IndexOutOfBoundsException if any of 2 conditions are true

Comment (V1talie) [reply src/main/java/com/endava/internship/collections/StudentList.java:148]: Done

**Review by GAlexandru-be — APPROVED** [review]

**Pull request merged**